drivers/gpio/gpio_sx1509b.c

```diff
@@ -320,12 +459,55 @@ static int pin_interrupt_configure(struct device *dev,
 				   enum gpio_int_mode mode,
 				   enum gpio_int_trig trig)
 {
-	int ret = 0;
+	const struct sx1509b_config *cfg = dev->config->config_info;
+	struct sx1509b_drv_data *drv_data = dev->driver_data;
+	struct sx1509b_pin_state *pins = &drv_data->pin_state;
+	int rc = 0;
 
-	if (mode != GPIO_INT_MODE_DISABLED) {
-		ret = -ENOTSUP;
+	if (!IS_ENABLED(CONFIG_GPIO_SX1509B_INTERRUPT) &&
+			(mode != GPIO_INT_MODE_DISABLED)) {
+		return -ENOTSUP;
 	}
-	return ret;
+
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	if (mode == GPIO_INT_MODE_EDGE) {
+		pins->interrupt &= ~BIT(pin);
```

Comment:
The `gpio_{en,dis}able_callback()` API, which also adjusts `cb_pins`, is obsolete, so as long as invoking the callback depends on a match between detected sense and enabled pins this function must also maintain `cb_pins`.